### PR TITLE
[12.0][IMP] helpdesk_stock: Improved code.

### PR DESCRIPTION
**Pull request by murtuzasaleh**
Improved code.

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
@max3903 @osi-scampbell Can you please review this PR.

File: helpdesk_stock/models/helpdesk_ticket.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ def _default_warehouse_id(self):
             [('company_id', '=', company)], limit=1)
         return warehouse_ids
 
+    inventory_location_id = fields.Many2one(
+        'stock.location', 'Destination Location')
```
Comment (max3903) [inline helpdesk_stock/models/helpdesk_ticket.py:26]: The destination location should be determined on the stock request, not on the ticket.

Comment (murtuzasaleh) [reply helpdesk_stock/models/helpdesk_ticket.py:26]: But it is already added before refactoring stock.request code.

File: helpdesk_stock/models/helpdesk_ticket.py [hunk 2]
```diff
@@ -48,6 +50,8 @@ def action_confirm(self):
         for rec in self:
             if not rec.stock_request_ids:
                 raise UserError(_('Please select a Materials.'))
+            if not self.warehouse_id and not self.warehouse_id:
```
Comment (max3903) [inline helpdesk_stock/models/helpdesk_ticket.py]:
```suggestion
            if not self.warehouse_id and not self.inventory_location_id:
```

**Review by max3903 — APPROVED** [review]

**Review by osi-scampbell — APPROVED** [review]

**Pull request merged**